SPARQLWrapper/main.py

```diff
@@ -136,18 +135,27 @@ def main(test=None):
     sparql.setReturnFormat(args.format)
     results = sparql.query().convert()
 
-    if args.format == "json":
+    ct = type(results)
+    if ct is dict:
+        # "json"
         print(json.dumps(results, indent=4))
-    elif args.format in ("xml", "rdf+xml", "json-ld"):
-        print(results.toxml())  # type: ignore
-    elif args.format == "n3":
-        g = rdflib.Graph()
-        g.parse(data=results, format="n3")  # type: ignore
-        print(g.serialize(format="n3"))
-    elif args.format in ("csv", "tsv", "turtle"):
-        print(results.decode("utf-8"))  # type: ignore
+    elif ct is xml.dom.minidom.Document:
+        # "xml", "rdf+xml", "json-ld"
+        print(results.toxml())
+    elif ct is bytes:
+        if args.format == "n3":
+            g = rdflib.Graph()
+            g.parse(data=results, format="n3")
+            print(g.serialize(format="n3"))
+        else:
+            # "csv", "tsv", "turtle"
+            print(results.decode("utf-8"))
+    elif ct is rdflib.graph.ConjunctiveGraph:
+        # "rdf"
+        print(results.serialize())
```

Comment:
Just a note, not something to change here: we should probably rationalize what convert does, as it seems quite random, and ultimately it seems like the requested response format will have limited impact on the actual response received.